x/ibc/src/ics02_client/types/client_state.rs

```diff
@@ -0,0 +1,66 @@
+use derive_more::{From, TryInto};
+//use gears::core::Protobuf;
+use gears::params::ParamsSubspaceKey;
+use gears::store::database::Database;
+use gears::store::StoreKey;
+
+use ibc::clients::tendermint::client_state::ClientState as TmClientState;
+use ibc::clients::tendermint::types::{
+    ClientState as ClientStateType, TENDERMINT_CLIENT_STATE_TYPE_URL,
+};
+use ibc::core::client::types::error::ClientError;
+use ibc::derive::ClientState;
+use ibc::primitives::proto::{Any, Protobuf};
+use serde::{Deserialize, Serialize};
+
+use crate::types::context::Context;
+
+#[derive(ClientState, Clone, From, TryInto, Debug, Serialize, PartialEq, Deserialize)]
+#[validation(Context<'a, 'b , DB: Database, SK: StoreKey, PSK:ParamsSubspaceKey >)]
+#[execution(Context<'a, 'b, DB: Database, SK:StoreKey, PSK: ParamsSubspaceKey>)]
+// TODO: this enum doesn't serialize to the same JSON as the Cosmos SDK. This is a separate issue to the derive macro issue.
+// E.g. durations are serialized as fields rather than strings, some fields are in camleCase rather than snake_case.
+//#[serde(tag = "@type")] // TODO: uncommenting this causes ClientState derive macro to panic. Uncomment when fixed
+pub enum ClientState {
```

Comment:
`#[serde(untagged)]`?

Reply:
No, it needs to be tagged to match Cosmos, unfortunately that causes that macro to panic.